contracts/src/v0.8/functions/dev/1_0_0/FunctionsSubscriptions.sol

```diff
@@ -0,0 +1,551 @@
+   * @dev Only callable by the Coordinator contract that is saved in the request commitment
+   * @param subscriptionId -
+   * @param estimatedCost -
+   * @param client -
+   * @param adminFee -
+   * @param juelsPerGas -
+   * @param gasUsed -
+   * @param costWithoutCallbackJuels -
+   */
+  function _pay(
+    uint64 subscriptionId,
+    uint96 estimatedCost,
+    address client,
+    uint96 adminFee,
+    uint96 juelsPerGas,
+    uint256 gasUsed,
+    uint96 costWithoutCallbackJuels
+  ) internal returns (Receipt memory receipt) {
+    uint96 callbackGasCostJuels = juelsPerGas * SafeCast.toUint96(gasUsed);
+    uint96 totalCostJuels;
+    totalCostJuels += costWithoutCallbackJuels;
+    totalCostJuels += adminFee;
+    totalCostJuels += callbackGasCostJuels;
+    receipt = Receipt(callbackGasCostJuels, totalCostJuels);
+
+    // Charge the subscription
+    s_subscriptions[subscriptionId].balance -= totalCostJuels;
+
+    // Pay the DON's fees and gas reimbursement
+    s_withdrawableTokens[msg.sender] += costWithoutCallbackJuels + callbackGasCostJuels;
+
+    // Pay out the administration fee
+    s_withdrawableTokens[address(this)] += adminFee;
+
+    s_subscriptions[subscriptionId].blockedBalance -= estimatedCost;
+    s_consumers[client][subscriptionId].completedRequests += 1;
+  }
+
+  // ================================================================
+  // |                      Owner methods                           |
+  // ================================================================
+  /**
+   * @inheritdoc IFunctionsSubscriptions
+   */
+  function ownerCancelSubscription(uint64 subscriptionId) external onlyRouterOwner {
+    address owner = s_subscriptions[subscriptionId].owner;
+    if (owner == address(0)) {
+      revert InvalidSubscription();
+    }
+    cancelSubscriptionHelper(subscriptionId, owner);
+  }
+
+  /**
+   * @inheritdoc IFunctionsSubscriptions
+   */
+  function recoverFunds(address to) external onlyRouterOwner {
+    uint256 externalBalance = LINK.balanceOf(address(this));
+    uint256 internalBalance = uint256(s_totalBalance);
+    if (internalBalance > externalBalance) {
+      revert BalanceInvariantViolated(internalBalance, externalBalance);
+    }
+    if (internalBalance < externalBalance) {
+      uint256 amount = externalBalance - internalBalance;
+      LINK.transfer(to, amount);
+      emit FundsRecovered(to, amount);
+    }
+    // If the balances are equal, nothing to be done.
+  }
+
+  /*
+   * @notice Owner withdraw LINK earned through admin fees
+   * @notice If amount is 0 the full balance will be withdrawn
+   * @param recipient where to send the funds
+   * @param amount amount to withdraw
+   */
+  function ownerWithdraw(address recipient, uint96 amount) internal nonReentrant onlyRouterOwner {
+    if (amount == 0) {
+      amount = s_withdrawableTokens[address(this)];
+    }
+    if (s_withdrawableTokens[address(this)] < amount) {
+      revert InsufficientSubscriptionBalance();
```

Comment:
The router owner (ie: CLL), is not a "subscription holder", right?  There isn't a subscription, just an account that can be withdrawn.  Not sure InsufficentSubscriptionBalance() makes sense as the error message.